localstack-core/localstack/services/stepfunctions/asl/utils/json_path.py

```diff
@@ -1,25 +1,37 @@
 import json
+import re
+from typing import Final
 
 from jsonpath_ng.ext import parse
 from jsonpath_ng.jsonpath import Index
 
 from localstack.services.stepfunctions.asl.utils.encoding import to_json_str
 
+_PATTERN_SINGLETON_ARRAY_ACCESS_OUTPUT: Final[str] = r"\[\d+\]$"
```

Comment:
Did an error case reveal this regex or is that our best guess of how the AWS heuristic works?

Reply:
The latter; this check appears to be restrictive enough for us to patch the output of the jsonpath library to match AWS's. However, this is indeed another sign that the jsonpath library does cover many of AWS SFN's logics